tools/torch_quant/torch_quant/module.py

```diff
@@ -22,11 +22,102 @@ class TracePair(NamedTuple):
 
 
 # eval priority: exclude > include, name > type
-class ModuleFilter(NamedTuple):
-    include_names: Optional[List[str]] = None
-    include_types: Optional[List[Type[nn.Module]]] = None
-    exclude_names: Optional[List[str]] = None
-    exclude_types: Optional[List[Type[nn.Module]]] = None
+class ModuleFilter:
+    # TODO(wanchen.swc): better names for *_classes, *_op_types arguments
+    def __init__(
+        self,
+        include_names: Optional[List[str]] = None,
+        include_classes: Optional[List[Type[nn.Module]]] = None,
+        include_op_types: Optional[List[Type[nn.Module]]] = None,
+        exclude_names: Optional[List[str]] = None,
+        exclude_classes: Optional[List[Type[nn.Module]]] = None,
+        exclude_op_types: Optional[List[Type[nn.Module]]] = None,
+    ):
+        """
+        Args:
+            include_names:
+                List of modules to quantize, identified by name.
+                If not None, only the modules in this list are quantized.
+                Example: ['foo.bar.name']
+            include_classes:
+                List of modules to quantize, identified by class (custom module).
+                If not None, only the modules in this list are quantized.
+                Example: ['MyModule']
+            include_op_types:
+                Specify the types of operators to quantize.
+                For [torch.nn.Conv2d], it will quantize torch.nn.Conv2d only.
+                It quantizes all supported operators by default.
+            exclude_names:
+                List of modules to exclude, identified by name.
+                If not None, modules in this list will be excluded from quantization.
+                Example: ['foo.bar.name']
+            exclude_classes:
+                List of modules to exclude, identified by class (custom module).
+                If not None, modules in this list will be excluded from quantization.
+                Example: ['MyModule']
+            exclude_op_types:
+                Specify the types of operators not to be quantized.
+                For [torch.nn.Conv2d], it will quantize all supported operators except
+                torch.nn.Conv2d.
+        """
+        self.include_names = include_names
+        self.include_classes = include_classes
+        self.include_op_types = include_op_types
+        self.exclude_names = exclude_names
+        self.exclude_classes = exclude_classes
+        self.exclude_op_types = exclude_op_types
+
+    def _submodule_names(self, names, module_name: str) -> Optional[List[str]]:
+        """
+        If module name is 'foo', turn full path 'foo.bar.name' into 'bar.name'
+        """
+        if names and module_name:
+            lstrip_func = lambda x : x.replace(f'{module_name}.', '', 1)
+            names = [lstrip_func(m) for m in names if m.startswith(f'{module_name}.')]
+        return names or None
+
+    def submodule_filter(self, module_name: str):
```

Comment:
nitpicking: `submodule_filter` could be defined as a free function instead of method. Because this function doesn't access any private field (`_submodule_names` doesn't need to be method for the same reason).

Reply:
`submodule_filter` needs object's attribute values ( i.e. include_names, include_classes, ...). You mean we can define a function, pass in `module_filter` as incoming parameter?
`_submodule_names` really doesn't need to be method, has be modified as an internal function.